python/lsst/pipe/tasks/assembleCoadd.py

```diff
@@ -605,3 +622,298 @@ def makeDataRefList(self, namespace):
+        dtype = float,
+        default = 0.65
+    )
+    minClipFootOverlapSingle = pexConfig.Field(
+        doc = "Minimum fractional overlap of clipped footprint with visit DETECTED to be " \
+            "clipped when only one visit overlaps",
+        dtype = float,
+        default = 0.5
+    )
+    minClipFootOverlapDouble = pexConfig.Field(
+        doc = "Minimum fractional overlap of clipped footprints with visit DETECTED to be " \
+            "clipped when two visits overlap",
+        dtype = float,
+        default = 0.45
+    )
+    maxClipFootOverlapDouble = pexConfig.Field(
+        doc = "Maximum fractional overlap of clipped footprints with visit DETECTED when " \
+            "considering two visits",
+        dtype = float,
+        default = 0.15
+    )
+    minBigOverlap = pexConfig.Field(
+        doc = "Minimum number of pixels in footprint to use DETECTED mask from the single visits " \
+            "when labeling clipped footprints",
+        dtype = int,
+        default = 100
+    )
+
+    def setDefaults(self):
+        pexConfig.Config.setDefaults(self)
+        self.clipDetection.reEstimateBackground = False
+        self.clipDetection.returnOriginalFootprints = False
+        self.clipDetection.thresholdPolarity = "both"
+        self.clipDetection.thresholdValue = 2
+        self.clipDetection.nSigmaToGrow = 4
+        self.clipDetection.minPixels = 4
+        self.clipDetection.isotropicGrow = True
+        self.clipDetection.thresholdType = "pixel_stdev"
+        self.sigmaClip = 1.5
+        self.clipIter = 3
+
+
+class SafeClipAssembleCoaddTask(AssembleCoaddTask):
+    """Assemble a coadd, being careful to clip areas with potential artifacts
+
+    This is done in a number of steps:
+        - identify potential regions that need to be clipped
+        - find how much overlap there is between each potential region and the DETECTED mask for each visit
+        - apply a cut based on the visit overlaps to determine the clipped regions
+        - for big areas do not use the region as determined from the coadd, but rather all overlapping
+          footprints from the visit DETECTED plane.  This helps in flagging low surface brightness regions.
+        - assemble the coadd, ignoring the clipped regions.
+
+    The cuts used to determine the clipped regions were found empirically by examining a lot of HSC data.
+    It is unlikely that they will generalize well to other data sets.  The current algorithm is limited to
+    identifying artifacts that are found in only one or two visits.  A different threshold is applied
+    depending on the number overlapping visits.  The current thresholds work well at removing a majority
+    of artifacts.
+
+    """
+    ConfigClass = SafeClipAssembleCoaddConfig
+    _DefaultName = "safeClipAssembleCoadd"
+
+    def __init__(self, *args, **kwargs):
+        super(SafeClipAssembleCoaddTask, self).__init__(*args, **kwargs)
```

Comment:
The [coding standards say](https://confluence.lsstcorp.org/display/LDMDG/Python+Coding+Standard#PythonCodingStandard-DesigningforInheritance):
> 'super' SHOULD NOT be used unless the author really understands the implications (e.g. in a well-understood  multiple inheritance hierarchy).
Do we really need it here?